api/discovery/discovery.proto

```diff
@@ -72,6 +72,28 @@ message QueryRequest {
 }
 
 message QueryResponse {
-  repeated google.protobuf.Value results = 1;
+  repeated Resource results = 1;
   string next_page_token = 2;
+}
+
+// Resource is a wrapper around google.protobuf.Value that is needed for
+// persistence reasons.
+message Resource {
+  // Id contains a unique ID for each resource. This is specific for the cloud
+  // provider this resource was gathered for and can for example be a resource
+  // URL.
+  string id = 1 [ (validate.rules).string.uuid = true ];
+  // CloudServiceId is the UUID for the cloud service to which this resource
+  // belongs to.
+  string cloud_service_id = 2 [ (validate.rules).string.min_len = 1 ];
+  // ResourceType contains a comma seperated string of resource types according
+  // to our ontology.
+  string resource_type = 3 [ (validate.rules).string.min_len = 1 ];
```

Comment:
Is there a reason not to use a list here as well as in the assessment result and call it resource_types? 

```
// Resource types
  repeated string resource_types = 8 [
    (tagger.tags) = "gorm:\"serializer:json\"",
    (validate.rules).repeated .min_items = 1
  ];
```

Reply:
Sort of, because we do a filter on the resource type. I think this may also be possible with the JSON type but I am not sure if this is possible in the same way with Postgres and sqlite. I will do a quick check, if this works.

Reply:
I did a check and it is a complete mess to check this in a JSON. Therefore, I would keep it as string